app/scripts/controllers/permissions/index.js

```diff
@@ -188,22 +192,29 @@ export class PermissionsController {
    */
   async legacyExposeAccounts (origin, accounts) {
 
+    const existingAccounts = await this.getAccounts(origin)
+
+    if (existingAccounts.length > 0) {
+      throw new Error(
+        'May not call legacyExposeAccounts on origin with exposed accounts.'
+      )
+    }
+
     const permissions = {
       eth_accounts: {},
     }
 
+    // validate and modify permissions as necessary
     await this.finalizePermissionsRequest(permissions, accounts)
 
-    let error
-    try {
-      await new Promise((resolve, reject) => {
-        this.permissions.grantNewPermissions(origin, permissions, {}, err => (err ? resolve() : reject(err)))
-      })
-    } catch (err) {
-      error = err
-    }
+    await new Promise((resolve, reject) => {
+      this.permissions.grantNewPermissions(
+        origin, permissions, {}, err => (err ? reject(err) : resolve())
+      )
+      // don't bother sending an accountsChanged notification for legacy dapps
+    })
+    .catch(error => {
```

Comment:
An `await` followed by a call to `catch`—can we continue to the old `try...catch` pattern?

Reply:
Yup. There was a bug there before, but it wasn't (go figure) related to that pattern